src/librustc_parse/parser/mod.rs

```diff
@@ -389,18 +368,6 @@ impl<'a> Parser<'a> {
         // Make parser point to the first token.
         parser.bump();
 
-        if let Some(directory) = directory {
-            parser.directory = directory;
-        } else if !parser.token.span.is_dummy() {
-            if let Some(FileName::Real(path)) =
-                &sess.source_map().lookup_char_pos(parser.token.span.lo()).file.unmapped_path
-            {
-                if let Some(directory_path) = path.parent() {
-                    parser.directory.path = directory_path.to_path_buf();
```

Comment:
This code may be the source of tomaka's problems with https://github.com/rust-lang/rust/pull/66364.

Reply:
cc @tomaka (would be interesting if we can test the old code that broke, after this PR lands)